src/coreclr/jit/llvm.cpp

```diff
@@ -114,12 +137,25 @@ void Llvm::llvmShutdown()
 //    Module.Verify(LLVMVerifierFailureAction.LLVMAbortProcessAction);
 }
 
+void cleanUp()
+{
+    if (_localsMap != nullptr)
+    {
+        // These structs are new'ed to they need to be delete'd
+        for (std::pair<unsigned int, struct LocalVar*> element : *_localsMap)
+        {
+            delete element.second;
```

Comment:
So, I am a bit puzzled by this `delete`. Usually we do not `delete` things in the Jit, neither do we have destructors. All allocations are arena-based and utilize the compiler's allocator (`compiler->getAllocator`), which has a no-op `free`.

All memory is then released when the function is compiled. Do we have a specific reason why we need (what I am guessing is) STL's `new` instead of compiler's here?

Reply:
That sounds good, I replace the `new` by using the allocator and I can delete the `delete`.   Will do, thanks.

Reply:
Expanding on the dynamic allocation bit...

Basically, we don't use STL in the Jit, and so we don't allocate dynamically with plain `new`. The benefit from that is we don't need to `delete`, the downside is we don't have the STL classes. It is up to you if you want to diverge from the Jit or not, I can see benefits both ways.